macos/README.md

```diff
@@ -5,28 +5,48 @@ By default it seems that launchd de-prioritizes processes as "background" tasks
 
 For our CI hosts we consider the jenkins agent process to be the most critical thing on the machine so we must give it special flags to override the default behavior.
 
-## Preparing the Jenkins agent property list
+## Swarm client or JNLP Agent
+
+Configurations for both [swarm client plugin](https://plugins.jenkins.io/swarm) and permanent agents connecting via JNLP are provided.
+Both are supported by ROS2 CI and the swarm client is recommended since deployment requires no changes to the Jenkins host.
+But if you have trouble with the swarm client you may wish to use the JNLP agent.
+
+## Preparing the Jenkins agent
+
+Create the directory `$HOME/jenkins-agent` if it does not already exist.
+
+If using the swarm client, download the swarm client jar from [this repository][swarm-repo] and place it in the ~/jenkins-agent directory.
+We're currently using version 3.8.
+
+For JNLP agents, the agent.jar is best downloaded from your running Jenkins. With wget installed:
+```
+cd ~/jenkins-agent && wget https://ci.ros2.org/jnlpJars/agent.jar
+```
 
-The agent plist was written when dosa was reimaged December 2017 and updated to use the Swarm client in Feb 2018.
-It expects that the directory `$HOME/jenkins-agent` exists and that it contains the Jenkins agent program `swarm-client-3.8.jar`.
-The swarm client, when combined with the [Swarm plugin](https://plugins.jenkins.io/swarm) on the buildfarm, allows for easier addition and removal of temporary Jenkins nodes.
-It can be downloaded from [this repository](https://repo.jenkins-ci.org/releases/org/jenkins-ci/plugins/swarm-client/).
 The agent must have at least Java JDK 8 installed.
+For best results use the same version as our Jenkins master, currently `1.8.0_162`.
```

Comment:
maybe replace this with how to infer the version instead so it's less likely to get outdated.

Reply:
Excellent point! Updated in 4772aaa

Reply:
cool thanks!